### PR TITLE
Stream and limit requested devices from the DB

### PR DESCRIPTION
**Pull request by nkly**
Fixes OTA-2385. No pressure on reviewing it, but it would be nice to merge by the end of the year :)

### CODE REVIEW

File: src/main/scala/com/advancedtelematic/campaigner/actor/CampaignScheduler.scala [hunk 1]
```diff
@@ -54,9 +57,10 @@ class CampaignScheduler(director: DirectorClient,
   def receive: Receive = {
     case NextBatch =>
       log.debug("Requesting next batch")
-      // TODO (OTA-2383) limit or stream device IDs from DB
-      campaigns.requestedDevices(campaign.id)
-        .map(deviceIds => BatchToSchedule(deviceIds.take(batchSize)))
+      campaigns.requestedDevicesStream(campaign.id)
+        .take(batchSize.toLong)
+        .runFold(Set.empty[DeviceId])(_ + _)
+        .map(BatchToSchedule(_))
```
Comment (houcros) [inline src/main/scala/com/advancedtelematic/campaigner/actor/CampaignScheduler.scala]: `.map(BatchToSchedule)`

File: src/main/scala/com/advancedtelematic/campaigner/actor/CampaignScheduler.scala [hunk 2]
```diff
@@ -54,9 +57,10 @@ class CampaignScheduler(director: DirectorClient,
   def receive: Receive = {
     case NextBatch =>
       log.debug("Requesting next batch")
-      // TODO (OTA-2383) limit or stream device IDs from DB
-      campaigns.requestedDevices(campaign.id)
-        .map(deviceIds => BatchToSchedule(deviceIds.take(batchSize)))
+      campaigns.requestedDevicesStream(campaign.id)
```
Comment (koshelev) [inline src/main/scala/com/advancedtelematic/campaigner/actor/CampaignScheduler.scala]: Is not some special statement/connection configuration required to configure a page size jdbc client should use? I can imagine default behaviour will be for driver fetch all the data and than Slick will turn it into a stream

Comment (nkly) [reply src/main/scala/com/advancedtelematic/campaigner/actor/CampaignScheduler.scala]: Hmm, though there is no mention of it in Akka Streams docs, it seems like you're right: https://mariadb.com/kb/en/library/about-mariadb-connector-j/#streaming-result-sets. Will fix it.

**Comment by nkly** [comment]
@koshelev Ah, nope, it's just a little cleanup of the things left undone after "Retry update" feature implementation.

**Pull request merged**